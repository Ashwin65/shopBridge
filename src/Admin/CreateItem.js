import React,{useState, useEffect} from "react";
import {useDispatch} from 'react-redux';
import Modal from '../Components/Modal';
import {  Button, Form } from 'antd';
import CreateItemForm from '../Components/CreateItemForm';
import  {createItem} from '../redux/actions/createItemAction'

const CreateItem = (props) => {

  const {isModalOpen = false, value ={}, isEdit=false, setModalState} = props
  

    

    const dispatch = useDispatch();

    const saveItemToStore = (data) =>{
      setModalState(false)        
      dispatch(createItem(data))

    }

    const closeModal = () => {
        setModalState(false)
     
    }


  const showModal = () => {
    setModalState(true)
  };



    return (<>
        <Button type="primary" onClick={showModal}>
        Add a Item
      </Button>
        <Modal modalVisible ={isModalOpen} closeModal = {closeModal} children= {<CreateItemForm closeModal={closeModal} saveItem = {saveItemToStore} value={value} isEdit={isEdit}/>}/>
        </>
    )
}

export default CreateItem;