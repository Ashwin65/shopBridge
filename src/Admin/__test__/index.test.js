import { mount, shallow } from 'enzyme';
import React from 'react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

import CreateItem from "../index";

const mockStore = configureMockStore();

const store = mockStore({});

let wrapper;

const setModalState = jest.fn()

beforeEach(() => {
    wrapper = shallow(<Provider store={store}><CreateItem/></Provider>);
});

describe('Create Item Page Snapshot', () => {
    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })
})