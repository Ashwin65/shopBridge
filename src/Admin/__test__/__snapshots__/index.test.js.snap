// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Item Page Snapshot renders correctly 1`] = `ShallowWrapper {}`;
